File: Programmers/Lv2/타겟 넘버.py
from collections import deque

def solution(numbers, target):
    answer = 0

    n = len(numbers)

    q = deque()
    q.append([numbers[0], 0])   # 값, idx
    q.append([-1 * numbers[0], 0])


    while q:
        now, idx = q.popleft()
        idx += 1

        if idx < n:
            q.append([now + numbers[idx], idx])
            q.append([now - numbers[idx], idx])
        else:
            if now == target:
                answer += 1

    return answer

print(solution([1,1,1,1,1], 3))

File: Programmers/Lv2/타겟 넘버.js
function solution(numbers, target) {
    var answer = 0;
    
    const dfs = (nums, idx, cur) => {
        if(idx == nums.length && cur == target){
            answer += 1;
            return
        } else if (idx == nums.length && cur !== target){
            return;
        }
        
        dfs(numbers, idx + 1, cur - numbers[idx]);
        dfs(numbers, idx + 1, cur + numbers[idx]);
    }
    
    dfs(numbers, 0, 0)
    
    return answer;
}